pkg/cloudprovider/vsphere/zones.go

```diff
@@ -90,6 +94,10 @@ func (z *zones) GetZoneByNodeName(ctx context.Context, nodeName k8stypes.NodeNam
 
 	zone := cloudprovider.Zone{}
 
+	if len(z.region) == 0 || len(z.zone) == 0 {
```

Comment:
Maybe for a future PR, should this be an && check? Do we support only setting regions?

Reply:
I believe both are required for out-of-tree. I know in-tree only requires one of them (forgot which one). I definitely have only ever tested when both the zone and region are set.